server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessageVaultRestoreTask.java

```diff
@@ -0,0 +1,174 @@
+
+    private final AtomicLong successfulRestoreCount;
+    private final AtomicLong errorRestoreCount;
+    private final AtomicLong failRemovingCount;
+
+    DeletedMessageVaultRestoreTask(DeletedMessageVault deletedMessageVault, User userToRestore, MailboxManager mailboxManager) {
+        this.deletedMessageVault = deletedMessageVault;
+        this.userToRestore = userToRestore;
+        this.mailboxManager = mailboxManager;
+        this.restoreMailbox = MailboxPath.forUser(userToRestore.asString(), RESTORE_MAILBOX_NAME);
+        this.session = getComponent(() -> mailboxManager.createSystemSession(userToRestore.asString()));
+        this.successfulRestoreCount = new AtomicLong();
+        this.errorRestoreCount = new AtomicLong();
+        this.failRemovingCount = new AtomicLong();
+    }
+
+    @Override
+    public Result run() {
+        MessageManager restoreMailboxManager = restoreMailboxManager();
+
+        return Flux.from(deletedMessageVault.search(userToRestore, Query.ALL)).toStream()
+            .map(deletedMessage -> moveToMailbox(restoreMailboxManager, deletedMessage))
+            .reduce(Result.COMPLETED, Task::combine);
+    }
+
+    private Result moveToMailbox(MessageManager restoreMailboxManager, DeletedMessage deletedMessage) {
+        Result appended = appendToMailbox(restoreMailboxManager, deletedMessage);
+
+        return Optional.of(appended)
+            .filter(appendedResult -> appendedResult.equals(Result.COMPLETED))
+            .map(appendedResult -> Task.combine(appendedResult, deleteInTheVault(deletedMessage)))
+            .orElse(Result.PARTIAL);
+    }
+
+    private Result deleteInTheVault(DeletedMessage deletedMessage) {
+        try {
+            Mono.from(deletedMessageVault.delete(userToRestore, deletedMessage.getMessageId())).block();
+            return Result.COMPLETED;
+        } catch (Exception anyException) {
+            LOGGER.error("deleting message {} in the vault got error", deletedMessage.getMessageId(), anyException);
+            this.failRemovingCount.incrementAndGet();
+            return Result.PARTIAL;
+        }
+    }
+
+    private Result appendToMailbox(MessageManager restoreMailboxManager, DeletedMessage deletedMessage) {
+        try {
+            restoreMailboxManager.appendMessage(AppendCommand.builder().build(deletedMessage.getContent().get()), session);
+            this.successfulRestoreCount.incrementAndGet();
+            return Result.COMPLETED;
+        } catch (MailboxException e) {
+            LOGGER.error("append message {} to restore mailbox {} didn't success",
+                deletedMessage.getMessageId(), restoreMailbox.asString(), e);
+
+            this.errorRestoreCount.incrementAndGet();
+            return Result.PARTIAL;
+        }
+    }
+
+    private MessageManager restoreMailboxManager() {
+        return getComponent(() -> Optional.of(mailboxManager.mailboxExists(restoreMailbox, session))
+            .filter(mailboxExist -> mailboxExist)
```

Comment:
I think that `.filter(Boolean::booleanValue)` is more readable.

Reply:
it had been refactored by classical if else